### DISCUSSION TITLE
WPML copy to relationship field

### DISCUSSION THREAD

**Discussion by DovWebmark** [pods-framework/pods#6595] [Ideas]
If I set WPML to copy fields when duplicating a post, it will not copy any field which is a relationship. Even if I manually add the row to wp_podsrel, it will not load it in the copied page, and will mark the field as being empty.

**Comment by jimtrue** [comment]
@DovWebmark Can you please provide more detail? When you brought this to our Slack Support, you provided more information and we need that logged here as well in order to replicate the problem:

1) What steps did you take to create the issue (including your configuration for WPML for this particular post-type and relationship). Since this is a conflict with another plugin, we need as many details as possible since we'll be engaging WPML for assistance with this one.

2) What was your expected outcome?

3) Screenshots can also be helpful. The more information you can give, the faster we can get someone working on this one.

Thanks!

**Comment by DovWebmark** [comment]
I have a pod with a relationship field to another pod, and it is a required field.
I want to make a language duplicate with the field already filled in.
If I set the field in WPML to 'copy', it should be filled in. It is not. Even if I then set the field manually and save, I get a message that the field is empty.
If I go into the table wp_podsrel, I see that the entry is missing for the new page for that field.
Even if I add it manually, when loading the page, it is marked as empty.

**Comment by jimtrue** [comment]
Do you have a 'default value' set for the field? The 'required setting' for the relationship sounds like it's conflicting or it's a timing issue for the point when the copy to WPML takes place. @JoryHogeveen Do you have much experience with the WPML configurations? I don't think PolyLang exhibits this particular problem.

**Comment by DovWebmark** [comment]
No default value

**Comment by jimtrue** [comment]
@srdjan-jcc Also mentioning you on this one to see if this is behavior you've seen in your testing.

**Comment by JoryHogeveen** [comment]
Hi @DovWebmark @jimtrue 

I believe this is because the meta sync/copy feature doesn't translate the post ID's. So the translated posts will get ID's from an untranslated post, and these get filtered out once use try to edit or view a post.

@jimtrue Polylang has the exact same problem and this is not something Polylang or WPML should fix. These plugins don't know the context of a meta value so it could just be a numeric value.
We should hook into the sync mechanism and make sure relationship ID's are translated.

A plugin I wrote for Polylang: https://github.com/JoryHogeveen/pods-polylang-sync-meta (check the `dev` branch!)
Could be ported to a WPML version quite easily I think.

**Comment by jimtrue** [comment]
@JoryHogeveen Does this mean our relationship field connections don't currently translate when you translate a post? I think I follow the issue from your description above, but I think we need @sc0ttkclark 's feedback in here on the best way to handle, especially with the changes being planned in 2.8

Also, is there currently a workaround for this condition?

**Comment by JoryHogeveen** [comment]
> Does this mean our relationship field connections don't currently translate when you translate a post?

Nope they don't since translating in theory is just adding a new post.

There is no actual workaround for this.
The only thing you can do is to stop syncing/copying metadata and just fill it for each translation again OR just fix the errors for each translation you add manually.

**Comment by JoryHogeveen** [comment]
Btw, Polylang related issues: #1851 #3773 #3774

**Comment by jimtrue** [comment]
@JoryHogeveen #3774 says it's invalid because it was caused by a synchronizations setting from PolyLang.  Is this not the case for that one?

**Comment by JoryHogeveen** [comment]
That issue is not relationship related but just plain text. So meta sync wouldn't be a adviced setting.
Though, with my extension I add an option for each field in which you can set whether to sync the value or not. In that case that user could use metasync feature from Polylang and define sync options for each Pod field separately.

**Comment by JoryHogeveen** [comment]
Same issue (sync meta fields): #2006

**Comment by nillde** [comment]
I ran into the same issue (wpml not synching relationships for different languages). Is there any chance this might be resolved in the future? Is there a workaround? 

It would be a shame not to be able to use relationships on a two-language page.

**Comment by JoryHogeveen** [comment]
Hi @nillde 
There are plans to create an addon for Pods Pro regarding this issue (already done for Polylang).
At the moment relationships still work for translations but they cannot be synched.

**Comment by nikolinenielsen** [comment]
How is this going? I'm having the same problem. :)

**Comment by JoryHogeveen** [comment]
Update on this issue.
We've had a talk with the integration developers at WPML. Pods 2.8 will include an improved version of our current integration for fetching data.
This part (copy fields) will most likely get a dedicated plugin.

In short, this issue is in our pipeline for Pods addons.

**Comment by verstecken** [comment]
You can´t believe how much I am waiting for this to work! :) I am struggling for weeks...

**Comment by verstecken** [comment]
If anyone comes by here, unchecking the WPML setting/checkbox "Adjust IDs for multilingual functionality" was the solution for me! @nillde @nikolinenielsen @DovWebmark 

Found it in this thread:
https://wordpress.org/support/topic/issue-with-wpml-and-pods-templates-since-2-8/

**Reply by trinermedia** [reply]
Like @codeclinic , I can't confirm that this completely solves the problem. Unless I am doing something wrong. Here's what I did:
1. Installed pods
2. Created a pod "Download" with a relationship field to link to a normal post.
3. Created a second pod to extend normal posts with a relationship field to link back to a download.
4. Created a pod template that contains an EACH loop for the relationship field "posts" (printing the permalink as a clickable link with the post_title as the label). I set this template in the pod "Download" under Auto-Template as a single template. 
5. Installed WPML
6. Made the following settings in WPML:
– «Adjust IDs for multilingual functionality» disabled
– Made the post type «Download» and all created custom fields translatable.
7. Created a download and a post and linked them by the relationship field.
8. Translated both the download and the post.

Result: When viewing the translated download in the second language, the linked post is displayed in the source language. So the field was copied, but the translation was not used.

@JoryHogeveen Can you confirm this? Is this the current state of how Pods and WPML working (not) together concerning relationship fields?

By the way: WMPL's media translation also does not work if you use a File / Image / Video field. The field is copied but the translation is not used here either.

**Comment by codeclinic** [comment]
> If anyone comes by here, unchecking the WPML setting/checkbox "Adjust IDs for multilingual functionality" was the solution for me! @nillde @nikolinenielsen @DovWebmark
> 
> Found it in this thread: https://wordpress.org/support/topic/issue-with-wpml-and-pods-templates-since-2-8/

Sadly that evidently doesn't work for all scenarios.

**Comment by pavodev** [comment]
Hi everyone,
Any news on this?

In 2023 with the latest versions of Pods and WPML I still have this issue. I created a Custom Post Type and a relationship with Media files, I use the API to fetch the data on a mobile application and I need the translation_id / trid fields. In previous versions of Pods these fields where included in the API response but since a couple of updates not anymore..
